frontend/lib/src/AppNode.ts

```diff
@@ -860,11 +860,16 @@ export class AppRoot {
   ): AppRoot {
     const existingNode = this.root.getIn(deltaPath)
 
-    // If we're replacing an existing Block, this new Block inherits
-    // the existing Block's children. This prevents existing widgets from
-    // having their values reset.
-    const children: AppNode[] =
-      existingNode instanceof BlockNode ? existingNode.children : []
+    // If we're replacing an existing Block of the same type, this new Block
+    // inherits the existing Block's children. This prevents existing widgets
```

Comment:
nit: maybe write it a bit more general since we also have non-widget elements that have state, e.g.:

This prevents existing child elements from being unmounted and having their state reset.